darts/models/gaussian_process_filter.py

```diff
@@ -55,14 +56,19 @@ def filter(self,
                                               'deterministic (observations).')
         super().filter(series)
 
-        times = series.time_index.values.reshape(-1, 1)
         values = series.values(copy=False)
+        if series.has_datetime_index:
+            times = np.arange(series.n_timesteps).reshape(-1, 1)
+        else:
+            times = series.time_index.values.reshape(-1, 1)
+
+        not_nan_mask = np.all(~np.isnan(values), axis=1)
```

Comment:
This is great. Actually in the future it could be a good idea to update `darts.utils.missing_values.fill_missing_values()`, so that it could optionally use GP as a method for filling the missing values.